examples/calendar/frontend/src/Profile/Profile.tsx

```diff
@@ -0,0 +1,196 @@
+			headers: {
+				Authorization: `Bearer ${accessToken}`
+			}
+		})).data;
+		return createAppointment;
+	}
+	const setTimeAvailAPI = async (availableTimes: SetBooking) => {
+		const setAvailTimes = (await axios.post(`${process.env.REACT_APP_BACKEND_URL}/time`, {availableTimes} ,{
+			headers: {
+				'Content-Type': 'application/json',
+				Authorization: `Bearer ${accessToken}`
+			}
+		})).data;
+		return setAvailTimes;
+	}
+	const timesInitial = () => {
+		const timeReturn = [];
+		var someDate = new Date(selectedDate);
+		someDate.setHours(8, 0, 0);
+		for (var i = 0; i < 60; i++)
+		{
+			const newTime = someDate.getTime() + i * 15 * 60 * 1000;
+			timeReturn.push({ startTime: new Date(newTime), checked: false, disabled: false });	
+		}
+		return timeReturn;
+	}
+	
+	const createAppointment = async () => {
+		let timeStarts:number[] = [];
+		let timeEnds:number[] = [];
+		checkedTimes.forEach((time: string) => {
+			timeStarts.push(parseInt(time));
+			let endTime = parseInt(time) + 15 * 60 * 1000;
+			timeEnds.push(endTime);
+		});
+		const booking: SetBooking = {
+			"timeStarts": timeStarts,
+			"timeEnds": timeEnds,
+			"status": "available",
+			"userId": userV.user?.id || 0,
+		}
+		await setTimeAvailAPI(booking);
+		const data = await createAppointmentAPI();
+		const secret_c = (data as any).appointment.secret;
+		setVisibleLink(true);
+		setLink(`${process.env.REACT_APP_FRONTEND_URL}/appointment/${secret_c}`)
+
+	}
+	const [selectedDate, setSelectedDate] = useState<Date>(new Date());
+	const [visibleLink, setVisibleLink] = useState<Boolean>(false);
+	const [times, setTimes] = useState<Time[]>(timesInitial())
+	const [userV, setUser] = useState<State>({
+		loading: false,
+		user: undefined,
+		username: ""
+	});
+	const [checkedTimes, setCheckedTimes] = useState<string[]>([]);
+	const [link, setLink] = useState<string>("text");
+	useEffect(() => { 
+		setTimes(timesInitial());
+	}, [selectedDate]);
+
+	useEffect(() => {
+		const { accessToken } = auth;
+		const {
+			payload: { id }
+		} = jwtDecode<JwtDecoded>(accessToken);
+		async function fetchUserAPI() {
+			const user = (await axios.get<User>(`${process.env.REACT_APP_BACKEND_URL}/users/user`, {
+				headers: {
+					Authorization: `Bearer ${accessToken}`
+				}
+			})).data;
+			setUser({ ...userV, user });
+			const booked = await getTimeAvailAPI(id);
+			booked.forEach((booking: any) => {				
+				const tempTimes = [...times];
+				var foundIndex = tempTimes.findIndex(x => parseInt((x.startTime.getTime()/1000).toString()) == (booked.timeStart/1000));
+				tempTimes[foundIndex] = { startTime: booking.startTime, checked: false, disabled: true };
+				setTimes(tempTimes);
+			});
+		};
+		fetchUserAPI();
+	}, []);
+
+	const { accessToken } = auth;
+
+	const {
+		payload: { publicAddress }
+	} = jwtDecode<JwtDecoded>(accessToken);
+
+	const { loading, user } = userV;
+
+	const username = user && user.username;
+
+	return (
+		<div className="Profile">
```

Comment:
quite same content is in another file too. Can this be extracted out, and used in different files

Reply:
Just two files, would have made sense if there were more than 2 instances.